src/main/java/rs/teslaris/core/service/DocumentPublicationService.java

```diff
@@ -0,0 +1,29 @@
+package rs.teslaris.core.service;
+
+import java.util.List;
+import org.springframework.stereotype.Service;
+import rs.teslaris.core.dto.document.DocumentFileDTO;
+import rs.teslaris.core.dto.document.JournalPublicationDTO;
+import rs.teslaris.core.dto.document.JournalPublicationResponseDTO;
+import rs.teslaris.core.model.document.Document;
+import rs.teslaris.core.model.document.JournalPublication;
+
+@Service
+public interface DocumentPublicationService {
```

Comment:
Should we introduce the hierarchy (Document, JournalPublication, ConferencePublication, Monograph, etc)? I mean it looks to me controllers will be separated, why not services? I am afraid the implementation might be quite a big one, if we keep this approach (one service for complete hierarchy).

Reply:
We can keep the common methods inside this service and make separate services to handle operations specific for that document type.